frontend/src/screens/GameScreen/gameScreen.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useEffect, useState } from "react";
+import { View, Text, StyleSheet } from "react-native";
+import { BLUE, GREY } from "../../constants/colors";
+import FirebaseInteractor from "../../firebase/firebaseInteractor";
+import { UID } from "../../models/types";
+import { durstenfeldShuffle } from "../../utils/utils";
+import { DraxView, DraxProvider } from "react-native-drax";
+import DroppableRow from "../../components/DroppableRow";
+import { useNavigation } from "@react-navigation/core";
+import { LoadingScreen } from "../../components/LoadingScreen";
+import PrimaryButton from "./PrimaryButton";
+import SecondaryButton from "./SecondaryButton";
+
+const fi = new FirebaseInteractor();
+
+interface MaybeWordPair {
+    english?: string;
+    turkish: string;
+    correctEnglishWord: string;
+}
+
+export interface GameScreenProps {
+  route: any;
+}
+
+export interface SpecificGameScreenProps {
+    route: any;
+    shuffleFunction: (words: any[]) => any[];
+    topScreenRenderFunction: (numCorrect: number, numTotal: number) => JSX.Element;
+    isPairing: boolean;
+}
+
+export default function GameScreen(props: SpecificGameScreenProps) {
+    const [englishWords, setEnglishWords] = useState<string[]>()
+    const [turkishWords, setTurkishWords] = useState<MaybeWordPair[]>()
```

Comment:
Could we rename this? This variable doesn't actually represent the Turkish words (which would be `string[]`), instead it represents a MaybePair object. I feel like giving it a more accurate name will make the overall code more readable.

I'm open to ideas for a new name, a potential idea of mine is `currentPairs`?